File: test/fixtures/typescript/calculator.ts
export class Calculator {
  private result: number;

  constructor() {
    this.result = 0;
  }

  add(a: number, b: number): number {
    return a + b;
  }

  subtract(a: number, b: number): number {
    return a - b;
  }

  multiply(a: number, b: number): number {
    return a * b;
  }

  divide(a: number, b: number): number {
    if (b === 0) {
      throw new Error("Division by zero");
    }
    return a / b;
  }
}

export function factorial(n: number): number {
  if (n <= 1) {
    return 1;
  }
  return n * factorial(n - 1);
}

const calculator = new Calculator();
console.log(calculator.add(2, 3));

File: test/fixtures/python/calculator.py
class Calculator:
    def __init__(self):
        self.result = 0

    def add(self, a: int, b: int) -> int:
        return a + b

    def subtract(self, a: int, b: int) -> int:
        return a - b

    def multiply(self, a: int, b: int) -> int:
        return a * b

    def divide(self, a: int, b: int) -> int:
        if b == 0:
            raise ValueError("Division by zero")
        return a / b


def factorial(n: int) -> int:
    if n <= 1:
        return 1
    return n * factorial(n - 1)


calculator = Calculator()
print(calculator.add(2, 3))
